storage-operator/pkg/salt/client.go

```diff
@@ -25,63 +35,202 @@ type Client struct {
 func NewClient(creds *Credential) *Client {
-	const SALT_API_PORT int = 4507 // As defined in master-99-metalk8s.conf
-
 	address := os.Getenv("METALK8S_SALT_MASTER_ADDRESS")
 	if address == "" {
-		address = "http://salt-master"
+		address = "http://salt-master:4507"
 	}
 
 	return &Client{
-		address: fmt.Sprintf("%s:%d", address, SALT_API_PORT),
+		address: address,
 		client:  &http.Client{},
 		creds:   creds,
 		token:   nil,
 		logger:  log.Log.WithName("salt_api"),
 	}
 }
 
-// Test function, will be removed later…
-func (self *Client) TestPing() (map[string]interface{}, error) {
-	payload := map[string]string{
-		"client": "local",
-		"tgt":    "*",
-		"fun":    "test.ping",
+// Spawn a job, asynchronously, to prepare the volume on the specified node.
+//
+// Arguments
+//     ctx:      the request context (used for cancellation)
+//     nodeName: name of the node where the volume will be
+//
+// Returns
+//     The Salt job ID.
+func (self *Client) PrepareVolume(
+	ctx context.Context, nodeName string,
+) (string, error) {
+	// Use rand_sleep to emulate slow operation for now
+	payload := map[string]interface{}{
+		"client": "local_async",
+		"tgt":    nodeName,
+		"fun":    "test.rand_sleep",
+	}
+
+	self.logger.Info("PrepareVolume")
+
+	ans, err := self.authenticatedRequest(ctx, "POST", "/", payload)
+	if err != nil {
+		return "", errors.Wrapf(
+			err, "PrepareVolume failed (target=%s)", nodeName,
+		)
+	}
+	// TODO(#1461): make this more robust.
+	result := ans["return"].([]interface{})[0].(map[string]interface{})
+	return result["jid"].(string), nil
+}
+
+// Spawn a job, asynchronously, to unprepare the volume on the specified node.
+//
+// Arguments
+//     ctx:      the request context (used for cancellation)
+//     nodeName: name of the node where the volume will be
+//
+// Returns
+//     The Salt job ID.
+func (self *Client) UnprepareVolume(
+	ctx context.Context, nodeName string,
+) (string, error) {
+	// Use rand_sleep to emulate slow operation for now
+	payload := map[string]interface{}{
+		"client": "local_async",
+		"tgt":    nodeName,
+		"fun":    "test.rand_sleep",
+	}
+
+	self.logger.Info("UnprepareVolume")
+
+	ans, err := self.authenticatedRequest(ctx, "POST", "/", payload)
+	if err != nil {
+		return "", errors.Wrapf(
+			err, "UnprepareVolume failed (target=%s)", nodeName,
+		)
+	}
+	// TODO(#1461): make this more robust.
+	result := ans["return"].([]interface{})[0].(map[string]interface{})
+	return result["jid"].(string), nil
+}
+
+// Poll the status of an asynchronous Salt job.
+//
+// Arguments
+//     ctx:      the request context (used for cancellation)
+//     jobId:    Salt job ID
+//     nodeName: node on which the job is executed
+//
+// Returns
+//     The result of the job if the execution is over, otherwise nil.
+func (self *Client) PollJob(
+	ctx context.Context, jobId string, nodeName string,
+) (map[string]interface{}, error) {
+	jobLogger := self.logger.WithValues("jobId", jobId)
```

Comment:
Logger values are of format `Foo.Bar`, so `Salt.JobID` or similar.

Reply:
Done